### PR TITLE
More distributed validation scripts

### PR DESCRIPTION
**Pull request by glwagner**
This PR adds another validation script for quick tests and such of distributed `NonhydrostaticModel`. We've been using the benchmark script, but that has additional features (and as a benchmark, is expensive to run). Hopefully these prove useful for fleshing out support for distributed cases.

Right now I've just added a simple 2D turbulence script.

TODO:
- [x] Add output to 2D turbulence script
- [ ] Add a 3D `NonhydrostaticModel` case
- [ ] Add `HydrostaticFreeSurfaceModel` case
- [ ] Add a script that runs into #2347 

cc @johnryantaylor

### CODE REVIEW

**Review by francispoulin — COMMENTED** [review]
Things are much cleaner than they were last year!

I could approve this but I see there are more boxes to check will wait.

File: validation/distributed_simulations/mpi_nonhydrostatic_two_dimensional_turbulence.jl [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+arch = MultiArch(CPU(); topology, ranks=(1, Nranks, 1))
+grid = RectilinearGrid(arch; topology, size=(Nx, Ny), halo=(3, 3), x=(0, 2π), y=(0, 2π))
+
+@info "Built $Nranks grids:"
+@show grid
+
+model = NonhydrostaticModel(; grid, advection=WENO5(), closure=ScalarDiffusivity(ν=1e-4, κ=1e-4))
+
+# This doesn't work?
+# ϵ(x, y, z) = 2rand() - 1 # ∈ (-1, 1)
```
[the remark this answers is not in the document]
Comment (glwagner) [reply validation/distributed_simulations/mpi_nonhydrostatic_two_dimensional_turbulence.jl:37]: Might be a dispatch error / bug? Haven’t had the chance to look.

**Review by navidcy — APPROVED** [review]

**Comment by navidcy** [comment]
@glwagner resolve conflicts and merge?

**Comment by glwagner** [comment]
Done!

**Comment by navidcy** [comment]
@glwagner, perhaps we merge (even with 1 task out of 4) so that PR doesn't get stale? Or are you planning to work on this?

**Comment by glwagner** [comment]
I think that's a good idea @navidcy

**Comment by glwagner** [comment]
Although, it looks like there are some changes to src and I'm not sure these are finished. If you take a look and can determine it's all good then we can merge ! Otherwise I should probably look and see.

**Comment by glwagner** [comment]
Tests pass so maybe all is fine...

**Pull request merged**